### ISSUE TITLE
Can not compile contract that use solidity error handling

### ISSUE THREAD

**Issue by haythem96** [sc-forks/solidity-coverage#657]
I have a contract that use the new Error messages, and I get the following error:
`Error in plugin solidity-coverage: Error: Could not instrument: core/Controller.sol. (Please verify solc can compile this file without errors.) missing 'constant' at 'InvalidOracleAddress' (19:6)`

**Comment by haythem96** [comment]
> Hi, could you link to a file which produces this error? Or provide a reproduction case?

```
pragma solidity >=0.8.0 <0.9.0;

error InvalidSomeAddress(address someAddress);

contract Reproduce {

    address public someAddress;

    function tryReproduce(address _add) external {
        if (_add == address(0)) revert InvalidSomeAddress(_add);

        someAddress = _add;
    }

}
```

Got this:

```
Error in plugin solidity-coverage: Error: Could not instrument: core/Reproduce.sol. (Please verify solc can compile this file without errors.) missing 'constant' at 'InvalidSomeAddress' (3:6)
```

**Comment by cgewecke** [comment]
@haythem96 This is fixed in the latest version `0.7.17` (courtesy of the latest solidity-parser/parser which supports this syntax).

Have added a unit test with your reproduction case, thanks for that!
